src/dmd/expression.d

```diff
@@ -2140,6 +2140,7 @@ extern (C++) class ThisExp : Expression
     {
         // Class `this` should be an rvalue; struct `this` should be an lvalue.
         return type.toBasetype().ty != Tclass;
+        //return true;
```

Comment:
This should be fine the way it is. Otherwise, the `this` class reference needs to live in memory (and would theoretically be mutable).